Source/Products.Repository/ProductRepository.cs

```diff
@@ -32,7 +32,7 @@ public async Task<Product> GetProduct(string id)
 
         public async Task<IEnumerable<Product>> GetProductsByName(string name)
         {
-            FilterDefinition<Product> filter = Builders<Product>.Filter.Eq(p => p.Name, name);
+            FilterDefinition<Product> filter = Builders<Product>.Filter.Regex("Name", name);
```

Comment:
iterpreting the regex can have consequences , we need to highlight that exposing this method should only happen on sanitized input